runtime/lua/vim/shared.lua

```diff
@@ -200,6 +200,10 @@ function vim.tbl_isempty(t)
   return next(t) == nil
 end
 
+local function can_merge(v)
+  return v and type(v) == "table" and #v == 0
```

Comment:
```suggestion
  return type(v) == "table" and #v == 0
```

`type(v) == 'table'` would already imply that `v ~= nil`

Reply:
that was me trying to debug a segmentation fault. turned out it was tressitter (parsers were built with different libs). will change again